### PR TITLE
Clean up the legacy MVC a bit more.

### PR DESCRIPTION
**Pull request by realityking**
Mostly deprecating some stuff or adding the log messages for deprecated stuff. Also adds a $input member to hold a JInput object to JControllerLegacy. This makes the old JController a bit more like the new one.

### CODE REVIEW

File: libraries/legacy/view/legacy.php [hunk 1]
```diff
@@ -546,9 +553,12 @@ public function setLayoutExt($value)
 	 * @return  void
 	 *
 	 * @since   12.2
+	 * @deprecated  13.3  Override JViewLeacy::escape() instead.
```
Comment (elinw) [inline libraries/legacy/view/legacy.php]: JViewLegacy?

File: libraries/legacy/view/legacy.php [hunk 2]
```diff
@@ -546,9 +553,12 @@ public function setLayoutExt($value)
...
 	 */
 	public function setEscape($spec)
 	{
+		JLog::add(__METHOD__ . ' is deprecated. Override JViewLeacy::escape() instead.', JLog::WARNING, 'deprecated');
+
```
Comment (elinw) [inline libraries/legacy/view/legacy.php]: JViewLegacy?

Comment (mbabker) [inline libraries/legacy/view/legacy.php]: The legacy JView class.  Renamed to deconflict with the new MVC JView interface.

Comment (realityking) [inline libraries/legacy/view/legacy.php]: I got a typo, it says JViewLeacy. I'll fix it in a bit.

Comment (realityking) [inline libraries/legacy/view/legacy.php]: Fixed.

**Comment by realityking** [comment]
Don't merge this just yet, legacy phpunit fails right now.

**Comment by realityking** [comment]
Ok, all clear :) Easier than I thought.

**Pull request merged**